core/src/main/scala/org/apache/spark/shuffle/sort/SortShuffleManager.scala

```diff
@@ -127,6 +138,7 @@ private[spark] class SortShuffleManager(conf: SparkConf) extends ShuffleManager
       mapId: Int,
       context: TaskContext,
       metrics: ShuffleWriteMetricsReporter): ShuffleWriter[K, V] = {
+    initializeExecutor()
```

Comment:
You dont' want to to do this at every `getWriter` call, but only once per executor. Since there's only one `SortShuffleManager` per executor, I think you can do it as a private variable in this class. You can get rid of the `initialized` flat then too.

Reply:
Agreed